internal/awsservice/cloudwatchlogs.go

```diff
@@ -134,30 +71,50 @@ func DeleteLogGroup(logGroupName string) {
 	}
 }
 
-// ValidateLogsInOrder takes a log group, log stream, a list of specific log lines and a timestamp.
-// It should query the given log stream for log events, and then confirm that the log lines that are
-// returned match the expected log lines. This also sanitizes the log lines from both the output and
-// the expected lines input to ensure that they don't diverge in JSON representation (" vs ')
-func ValidateLogsInOrder(t *testing.T, logGroup, logStream string, logLines []string, since time.Time) {
-	log.Printf("Checking %s/%s since %s for %d expected logs", logGroup, logStream, since.UTC().Format(time.RFC3339), len(logLines))
-	cwlClient, clientContext, err := getCloudWatchLogsClient()
+// ValidateLogs queries a given LogGroup/LogStream combination given the start and end times, and executes an
+// arbitrary validator function on the found logs.
+func ValidateLogs(logGroup, logStream string, since, until *time.Time, validator func(logs []string) bool) (bool, error) {
```

Comment:
Would it be better to change to `startTime` and `endTime`  instead of since and until?

Reply:
StartTime and EndTime are the field attributes in the API call / SDK, but that doesn't mean we have to change it here. If you feel strongly about it, I can change it